framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -162,6 +165,7 @@ public boolean usesSequentialSemantics() {
      * @param store the store
      * @return the possibly-modified value
      */
+    @Pure
```

Comment:
Shouldn't this, and the other `finishX` method below, stay only SideEffectFree? To allow a subclass to return a new object?

Reply:
Okay, I forgot to consider about overriding.